src/content/docs/en/guides/middleware.mdx

```diff
@@ -196,6 +197,76 @@ auth response
 validation response
 ```
 
+## Rewriting
+
+<Since v="4.13.0" />
+
+The `APIContext` exposes a method called `rewrite`, which works the same way as [Astro.rewrite](/en/guides/routing/#rewrites).
+
+```js title="src/middleware.js"
+import { isLoggedIn } from "~/auth.js"
+export function onRequest (context, next) {
+  if (!isLoggedIn(context) {
+    // If the user is not logged in, update the Request to render the `/login` route and
+    // add header to indicate where the user should be sent after a successful login.
+    return context.rewrite(new Request("/login", {
+      headers: {
+        "x-redirect-to": context.url.pathname
+      }
+    }));
+  }
+
+  return next();
+};
+```
+
+The `next` function optionally accept a parameter that allows to **rewrite** the current `Request`. For example, if a user isn't logged in, you could **rewrite** the current request without the need of a redirect:
```

Comment:
I'm not sure what the difference between these two slightly different methods is, practically speaking. Do they both do the exact same thing in the end?

If we are going to show both, is the point to say that you can "do the same thing either way"? Is the second one a "convenient shortcut" or preferable in any way? These examples look very similar on first glance, and someone might have trouble seeing the difference without looking closely. So it's helpful if we can draw the reader's attention to why we are showing them two slightly different examples, and what we want them to learn from them.

And, if you expect people to use only ONE of them 80% of the time, it's often OK to only show the common/preferred one. If they both have their place/advantages, then it's helpful if we explain why when we show both.

Reply:
Ok, I think I see the confusion, probably because I put the important bits in the incorrect context. Here's the breakdown:

`ctx.rewrite`:
- it retriggers a new rendering phase (the middleware is called again)
- it doesn't call subsequent middleware functions, in case `sequence` is used

`next()` with payload
- it doesn't retrigger a new rendering phase
- it calls subsequent middleware functions, in case `sequence` is used. The next middleware functions will receive the new `Request`

Reply:
Thanks! This is helpful! I will look at this more closely after our calls!

Reply:
Ok, @ematipico - Please check [this commit](https://github.com/withastro/docs/pull/8914/commits/7dc8649f13631385971dc0d15842ce48798dd243) to see if this might be an accurate way to distinguish the two samples.

Since they do in fact act differently but the code comments are identical, I added a third `//` code comment line which should be different to help illustrate how the two examples work differently, or when you might use one instead of the other, or what the different effects are.

I also edited the intro paragraph to each one a tiny bit, but I think if we get that third code comment line just right, then this will be really helpful! So please check those in particular and correct them as necessary!